java-squid/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -324,11 +351,17 @@ public Scope typeParameters() {
     }
 
     public String getFullyQualifiedName() {
-      String ownerName = "";
-      if (!owner.name.isEmpty()) {
-        ownerName = owner.name + ".";
+      String newQualification;
+      if (owner.isPackageSymbol()) {
+        newQualification = owner.name.isEmpty() ? "" : owner.name + ".";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)